clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -3919,18 +3919,16 @@ void CodeGenFunction::EmitTrapCheck(llvm::Value *Checked,
     Builder.CreateCondBr(Checked, Cont, TrapBB);
     EmitBlock(TrapBB);
 
-    llvm::CallInst *TrapCall = Builder.CreateCall(
-        CGM.getIntrinsic(llvm::Intrinsic::ubsantrap),
-        llvm::ConstantInt::get(CGM.Int8Ty,
-                               ClSanitizeDebugDeoptimization
-                                   ? TrapBB->getParent()->size()
-                                   : static_cast<uint64_t>(CheckHandlerID)));
+    llvm::CallInst *TrapCall =
+        Builder.CreateCall(CGM.getIntrinsic(llvm::Intrinsic::ubsantrap),
+                           llvm::ConstantInt::get(CGM.Int8Ty, CheckHandlerID));
 
     if (!CGM.getCodeGenOpts().TrapFuncName.empty()) {
       auto A = llvm::Attribute::get(getLLVMContext(), "trap-func-name",
                                     CGM.getCodeGenOpts().TrapFuncName);
       TrapCall->addFnAttr(A);
     }
+    TrapCall->addFnAttr(llvm::Attribute::NoMerge);
```

Comment:
Doesn't line 3918 (`TrapBB = createBasicBlock("trap");`) guarantee that `TrapBB != nullptr`?

Reply:
I am saying that we get into this "branch" when TrapBB was nullptr, the first time we see this kind of trap.

Reply:
Ah, I see, thanks. I've fixed it (and the test) in https://github.com/llvm/llvm-project/pull/117651/commits/5107ce194c9b03c2f32eb1508df0c92b13b3e2fd

Reply:
https://github.com/llvm/llvm-project/commit/5107ce194c9b03c2f32eb1508df0c92b13b3e2fd doesn't work because of inlining: none of the ubsantraps in function m() have nomerge, because they were each unique ubsantraps in the non-inlined functions.

Is there a downside to emitting nomerge for every ubsantrap (even the first instance)?

Reply:
Thanks, fixed in https://github.com/llvm/llvm-project/pull/117651/commits/0e8a4143ddc7b5c6e9ca7e64729ba5adcb0d92fb